edb/schema/types.py

```diff
@@ -1359,7 +1362,13 @@ def _to_nonpolymorphic(
         schema: s_schema.Schema,
         concrete_type: Type,
     ) -> typing.Tuple[s_schema.Schema, Array]:
-        return Array.from_subtypes(schema, (concrete_type,))
+        st = self.get_subtypes(schema=schema)[0]
```

Comment:
It sketches me out to have specific range handling here instead of a more generalized system for resolving nested polymorphic things.
Add a TODO comment about fixing it, at least?

Reply:
Added a TODO